src/base/configuration.cpp

```diff
@@ -110,8 +110,105 @@ std::string Configuration::getVersionExtra() {
   return CVC4_EXTRAVERSION;
 }
 
+std::string Configuration::copyright() {
+  std::stringstream ss;
+  ss << "Copyright (c) 2009-2017 by the authors and their institutional\n"
+     << "affiliations listed at http://cvc4.cs.stanford.edu/authors\n\n";
+
+  if (Configuration::licenseIsGpl()) {
+    ss << "This build of CVC4 uses GPLed libraries, and is thus covered by\n"
+       << "the GNU General Public License (GPL) version 3.  Versions of CVC4\n"
+       << "are available that are covered by the (modified) BSD license. If\n"
+       << "you want to license CVC4 under this license, please configure CVC4\n"
+       << "with the \"--bsd\" option before building from sources.\n\n";
+  } else {
+    ss << "CVC4 is open-source and is covered by the BSD license (modified)."
+       << "\n\n";
+  }
+
+  ss << "THIS SOFTWARE IS PROVIDED AS-IS, WITHOUT ANY WARRANTIES.\n"
+     << "USE AT YOUR OWN RISK.\n\n";
+ 
+  ss << "This version of CVC4 is linked against the following third party "
+     << "libraries:\n\n";
```

Comment:
I'm a bit concerned about including copyright dates for the third party libraries as these may change at any time and may vary from system to system.  What if we just list the copyrights without dates?

Reply:
I think that we can't. It usually is always required that the copyright notice is left intact (and this is in particular true for (L)GPL code) and I already bent that rule a little since I use ranges of years, e.g., 2000-2014, rather than a list of years (as is the case in the original copyright notice for CLN).